_posts/2017-12-19-2017-year-in-review.md

```diff
@@ -0,0 +1,144 @@
+
+Next, let's take a look at a few other project stats, starting with number of commits. If we run:
+
+```bash
+git log --pretty=oneline --after=12/31/2016 | wc -l
+```
+
+...inside the Flink repository, we'll see a total of **2316** commits so far in 2017, bringing the all-time total commits to **12,532**.
+
+Now, let's go a bit deeper. And here are instructions in case you'd like to take a look at this data yourself.
+
+Download gitstats from the [project homepage](http://gitstats.sourceforge.net/). Or, on OS X with homebrew, type:
+
+<!-- TODO: No longer works, any alternatives? -->
+
+```bash
+brew install --HEAD homebrew/head-only/gitstats
+```
+
+Clone the Apache Flink git repository:
+
+```bash
+git clone git@github.com:apache/flink.git
+```
+
+Generate the statistics
+
+```bash
+gitstats flink/ flink-stats/
+```
+
+View all the statistics as an html page using your default browser:
+
+```bash
+open flink-stats/index.html
+```
+
+Flink surpassed 1 million lines of code in 2016, and that trend continued in 2017 with the code base now clocking in at **1,257,949** lines.
+
+![Flink Total Lines of Code]({{ site.baseurl }}/img/blog/flink-lines-of-code-2017.png)
+
+Monday remains the day of the week with the most commits over the project's history, but Wednesday is catching up:
+
+![Flink Commits by Day of Week]({{ site.baseurl }}/img/blog/flink-dow-2017.png)
+
+And 5pm remains the preferred commit time, closely followed by 4pm:
+
+![Flink Commits by Hour of Day]({{ site.baseurl }}/img/blog/flink-hod-2017.png)
+
+### Meetups
+
+[Apache Flink Meetup membership](https://www.meetup.com/topics/apache-flink/) grew by **20%** this year, and at the time of writing, the number of meetups fell by 2 to **39**, but membership rose to **19,767** members listing Flink as a topic. With meetups in five continents, the Flink community is proud to be truly global in nature.
+
+![Apache Flink Meetup Map]({{ site.baseurl }}/img/blog/flink-meetups-dec2017.png)
+
+## Flink Forward 2017
+
+2017 was the first year we ran a Flink Forward conference in both [Berlin](https://berlin-2017.flink-forward.org) (September 11-13) and [San Francisco](https://sf-2017.flink-forward.org) (April 10-11), and over 350 members of our community attended each event for speaker sessions, training, and discussion about Flink.
+
+[Slides](https://www.slideshare.net/FlinkForward/presentations) and [videos](https://www.youtube.com/channel/UCY8_lgiZLZErZPF47a2hXMA) are available for all speaker sessions, and if you're interested in learning more about how organizations use Flink in production we encourage you to browse and watch a couple.
+
+For 2018, Flink Forward will be back in [September in Berlin](https://flink-forward.org/), and in [April in San Francisco](https://sf-2018.flink-forward.org/).
+
+![Flink Forward Speakers]({{ site.baseurl }}/img/blog/speaker-logos-ff2017.png)
+
+## Features and Ecosystem
+
+### Stateful Stream Processing with dA Platform 2
```

Comment:
@wints I was unsure if this should be mentioned or not…

Reply:
@ChrisChinchilla No, that shouldn't be included here.